src/DotNetWorker/Context/Features/DefaultModelBindingFeature.cs

```diff
@@ -38,12 +38,17 @@ public DefaultModelBindingFeature(IEnumerable<IConverter> converters)
                 FunctionParameter param = context.FunctionDefinition.Parameters[i];
 
                 object? source = context.Invocation.ValueProvider.GetValue(param.Name, context);
-                var converterContext = new DefaultConverterContext(param, source, context);
-
-                if (TryConvert(converterContext, out object? target))
+                if (source is not null)
                 {
-                    _parameterValues[i] = target;
-                    continue;
+                    var converterContext = new DefaultConverterContext(param, source, context);
+                    if (TryConvert(converterContext, out object? target))
+                    {
+                        _parameterValues[i] = target;
+                    }
+                    else
+                    {
+                        throw new InvalidCastException($"Cannot convert input parameter '{param.Name}' for Function '{context.FunctionDefinition.Name}' to type '{param.Type.FullName}' from type '{source.GetType().FullName}'.");
+                    }
```

Comment:
I'd like to err on the side of warning here rather than throwing. I can't think of any right now, but there may be valid scenarios to let null flow through to the function. And that way, a function author can handle null in whatever way they want (maybe they just ignore it).

Be sure to use `WorkerMessage.Define` so we can see how often this happens in backend logs.